### ISSUE TITLE
`AzureStaticWebApp@0` is broken

### ISSUE THREAD

**Issue by jpzwarte** [Azure/static-web-apps#828]
~~TLDR: this seems be since `0.203.0` was released. Pipelines with previous versions deployed fine.~~
TLDR: problem is probably in the Docker image used by the `AzureStaticWebApp` task.

This is the task in question from `azure-pipelines.yaml`:

```yaml
  - task: AzureStaticWebApp@0
    displayName: 'Publish preview docs'
    inputs:
      app_location: 'dist'
      deployment_environment: 'pullrequest'
      output_location: ''
      skip_app_build: true
      verbose: true
    env:
      azure_static_web_apps_api_token: $(deployment_token)
    condition: and(succeeded(), ne(variables.isMain, true))
```

I have tried everything. This configuration *should* work afaict (setting `output_location: ''` is in the docs).

First, I have added a debug task that does `ls -l dist/index.html`. The output from that is:

```
Starting: Debug
==============================================================================
Task         : Command line
Description  : Run a command line script using Bash on Linux and macOS and cmd.exe on Windows
Version      : 2.201.1
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line
==============================================================================
Generating script.
Script contents:
ls -l dist/index.html
========================== Starting Command Output ===========================
/bin/bash --noprofile --norc /tfs/_work/_temp/1692cfbe-8001-4c3d-b64c-f338bf25c939.sh
-rw-r--r-- 1 root root 2303 Jun 16 11:49 dist/index.html
Finishing: Debug
```

So `dist/index.html` *EXISTS*.

Next, the output from the `AzureStaticWebApp@0` task:

```
Starting: Publish preview docs
==============================================================================
Task         : Deploy Azure Static Web App
Description  : [PREVIEW] Build and deploy an Azure Static Web App
Version      : 0.203.0
Author       : Microsoft Corporation
Help         : https://aka.ms/swadocs
==============================================================================
/bin/bash /tfs/_work/_tasks/AzureStaticWebApp_18aad896-e191-4720-88d6-8ced4806941a/0.203.0/launch-docker.sh
stable: Pulling from appsvc/staticappsclient
Digest: sha256:c1fac5b3c048bc93533acf15079a926497d091e6a9e961e7ac878eb2679fe677
Status: Image is up to date for mcr.microsoft.com/appsvc/staticappsclient:stable
DeploymentId: 3f01ef13-87c6-44de-9c59-cd584a11eeca

Verbose logging enabled
Build timeout not specified, defaulting to 15 minutes
App Directory Location: 'dist' was found.
No Api directory specified. Azure Functions will not be created.
Looking for event info
Event info parsed from action options.
Skipping step to build /working_dir/dist with Oryx
Could not find staticwebapp.config.json file.
Didn't find Oryx manifest file under location: /ss-oryx/app-manifest/oryx-manifest.toml
Failed to find a default file in the app artifacts folder (dist). Valid default files: index.html,Index.html.
If your application contains purely static content, please verify that the variable 'app_location' in your deployment configuration file points to the root of your application.
If your application requires build steps, please validate that a default file exists in the build output directory.


For further information, please visit the Azure Static Web Apps documentation at https://docs.microsoft.com/en-us/azure/static-web-apps/
If you believe this behavior is unexpected, please raise a GitHub issue at https://github.com/azure/static-web-apps/issues/
Exiting
##[error]Error: The process '/bin/bash' failed with exit code 1
Finishing: Publish preview docs
```

I have also tried adding a dummy `staticwebapp.config.json` (with just `{}` as contents) to see if that helps (after reading #710), but even after setting `config_file_location: '/'`, it still says it cannot find that file:

```
config_file_location: config file '/staticwebapp.config.json' could not be found.
```

I have also tried setting `is_static_export: true` to see if that helps, but it doesn't.

I have tried adding a dummy build command and setting `app_location: '/'` and `output_location: 'dist'`, but that doesn't help.

This is pretty serious since it prevents us currently from deploying to production (which is not this task, but a different one with pretty much the same config).

**Comment by jpzwarte** [comment]
@rupareddy5-21 Any ideas perhaps?

**Comment by jpzwarte** [comment]
So just to provide as much information as possible, this is the dummy build config:

```yaml
  - task: AzureStaticWebApp@0
    displayName: 'Publish preview docs'
    inputs:
      app_build_command: 'echo 1'
      app_location: '/'
      deployment_environment: 'pullrequest'
      output_location: 'dist'
      verbose: true
    env:
      azure_static_web_apps_api_token: $(deployment_token)
    condition: and(succeeded(), ne(variables.isMain, true))
```

And the output:

```
Starting: Publish preview docs
==============================================================================
Task         : Deploy Azure Static Web App
Description  : [PREVIEW] Build and deploy an Azure Static Web App
Version      : 0.203.0
Author       : Microsoft Corporation
Help         : https://aka.ms/swadocs
==============================================================================
/bin/bash /tfs/_work/_tasks/AzureStaticWebApp_18aad896-e191-4720-88d6-8ced4806941a/0.203.0/launch-docker.sh
stable: Pulling from appsvc/staticappsclient
Digest: sha256:c1fac5b3c048bc93533acf15079a926497d091e6a9e961e7ac878eb2679fe677
Status: Image is up to date for mcr.microsoft.com/appsvc/staticappsclient:stable
DeploymentId: 148c4ab8-d4ba-40cb-90ca-3105b3789ff0

Verbose logging enabled
Build timeout not specified, defaulting to 15 minutes
App Directory Location: '/' was found.
No Api directory specified. Azure Functions will not be created.
Looking for event info
Event info parsed from action options.
Starting to build app with Oryx
Azure Static Web Apps utilizes Oryx to build both static applications and Azure Functions. You can find more details on Oryx here: https://github.com/microsoft/Oryx
Oryx will build app with the following custom override command: echo 1
Calling oryx build "/working_dir" -o /ss-oryx/app --manifest-dir /ss-oryx/app-manifest -p require_build -p disable_recursive_lookup=true -p ORYX_APP_TYPE="static-sites"
---Oryx build logs---


Operation performed by Microsoft Oryx, https://github.com/Microsoft/Oryx
You can report issues at https://github.com/Microsoft/Oryx/issues

Oryx Version: 0.2.20220401.1, Commit: afed4f696a08bfd4c8e4109704a082bf9fb20515, ReleaseTagName: 20220401.1

Build Operation ID: |u9b+xxtvZRE=.9bb96867_

Detecting platforms...
Could not detect any platform in the source directory.
Error: Could not detect the language from repo.


---End of Oryx build logs---
Oryx build has completed. Took 4.1030984 seconds. Exit Code: 2
Oryx was unable to determine the build steps. Continuing assuming the assets in this folder are already built. If this is an unexpected behavior please contact support.
Copying unbuilt assets from /working_dir to oryx app build folder /ss-oryx/app
Copied unbuilt assets from /working_dir to oryx app build folder /ss-oryx/app
Finished building app with Oryx
Could not find staticwebapp.config.json file.
Didn't find Oryx manifest file under location: /ss-oryx/app-manifest/oryx-manifest.toml
Failed to find a default file in the app artifacts folder (/). Valid default files: index.html,Index.html.
If your application contains purely static content, please verify that the variable 'app_location' in your deployment configuration file points to the root of your application.
If your application requires build steps, please validate that a default file exists in the build output directory.


For further information, please visit the Azure Static Web Apps documentation at https://docs.microsoft.com/en-us/azure/static-web-apps/
If you believe this behavior is unexpected, please raise a GitHub issue at https://github.com/azure/static-web-apps/issues/
Exiting
##[error]Error: The process '/bin/bash' failed with exit code 1
Finishing: Publish preview docs
```

**Comment by jpzwarte** [comment]
Using an older version of the task `- task: AzureStaticWebApp@0.202.0` also doesn't work. The problem is probably in the Docker image used.

**Comment by jpzwarte** [comment]
Also not because we use a custom Build Agent. Same issues with 

```yaml
pool:
  vmImage: ubuntu-latest
```

**Comment by rupareddy5-21** [comment]
Hey @jpzwarte Could you please share your repo so that we could try repro the same on our end. Thanks

**Comment by jpzwarte** [comment]
@rupareddy5-21 Unfortunately I can't: the repo is private. I was hoping by providing as much information as possible that you would have enough to reproduce the issue?

**Comment by jpzwarte** [comment]
@rupareddy5-21 If you *really* need access, then perhaps we can arrange something. In that case let's have contact over Teams?

**Comment by rupareddy5-21** [comment]
@jpzwarte That'd be great!

**Comment by jpzwarte** [comment]
@rupareddy5-21 I'm jeroen.zwartepoorte at sanoma.com on Teams.

**Comment by jpzwarte** [comment]
So after debugging, the issue lies with our custom docker image. Closing this issue as `AzureStaticWebApp` is working ok.
